File: InstagramRemakeJava/app/src/main/java/com/silvandrade/instagramremakejava/main/presentation/MainActivity.java
package com.silvandrade.instagramremakejava.main.presentation;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.drawable.Drawable;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import androidx.appcompat.widget.Toolbar;


import com.silvandrade.instagramremakejava.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
        }

        Toolbar toolbar = findViewById(R.id.main_toolbar);
        setSupportActionBar(toolbar);

        if(getSupportActionBar() != null) { // Definindo o item da câmera na toolbar.
            Drawable drawable = getResources().getDrawable(R.drawable.ic_insta_camera);
            getSupportActionBar().setHomeAsUpIndicator(drawable);
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }
    }
}

File: Instagram/app/src/main/java/co/tiagoaguiar/course/instagram/main/view/MainActivity.kt
package co.tiagoaguiar.course.instagram.main.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.WindowInsetsController
import androidx.appcompat.widget.Toolbar
import androidx.core.content.ContextCompat
import co.tiagoaguiar.course.instagram.R

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Alterando as cores da statusbar nessa activity
        window.statusBarColor = ContextCompat.getColor(this, R.color.gray)
        window.insetsController?.setSystemBarsAppearance(
            WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS,
            WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS)

        val toolbar = findViewById<Toolbar>(R.id.main_toolbar)
        setSupportActionBar(toolbar) // Informando ao sistema que essa activity possui uma toolbar e essa toolbar será responsável por escutar eventos de ação.

        // Atribuindo icon na toolbar.
        supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_insta_camera) // icon
        supportActionBar?.setDisplayHomeAsUpEnabled(true) // clicável
        supportActionBar?.title = ""
    }
}